engine/gamesys/src/gamesys/resources/res_font.cpp

```diff
@@ -418,6 +421,28 @@ namespace dmGameSystem
         glyph->m_DataImageChannels = inglyph->m_Channels;
         font->m_DynamicGlyphs.Put(codepoint, glyph);
 
+        uint32_t prev_width, prev_height, prev_ascent;
+        dmRender::GetFontMapCacheSize(font->m_FontMap, &prev_width, &prev_height, &prev_ascent);
+
+        if (!font->m_IsDynamic)
+        {
+            font->m_IsDynamic = 1;
+
+            // Discard any precalculated size
+            prev_width = prev_height = prev_ascent = 0;
+        }
+
+        bool dirty = inglyph->m_Width > prev_width ||
+                      inglyph->m_Height > prev_height ||
+                      inglyph->m_Ascent > prev_ascent;
+        if (dirty)
+        {
+            uint32_t cell_width = dmMath::Max((uint32_t)inglyph->m_Width, prev_width);
+            uint32_t cell_height = dmMath::Max((uint32_t)inglyph->m_Height, prev_height);
+            uint32_t cell_ascent = dmMath::Max((uint32_t)inglyph->m_Ascent, prev_ascent);
+            dmRender::SetFontMapCacheSize(font->m_FontMap, cell_width, cell_height, cell_ascent);
+        }
```

Comment:
Essentially the bug fix. When first adding a dynamic glyph, we reset it to zero, and after that, we increase the size when we notice a larger size.